alf/networks/actor_networks.py

```diff
@@ -0,0 +1,242 @@
+                `InputPreprocessor`, each of which will be applied to the
+                corresponding input. If not None, then it must
+                have the same structure with `input_tensor_spec` (after reshaping).
+                If any element is None, then it will be treated as alf.layers.identity.
+                This arg is helpful if you want to have separate preprocessings
+                for different inputs by configuring a gin file without changing
+                the code. For example, embedding a discrete input before concatenating
+                it to another continuous vector.
+            preprocessing_combiner (NestCombiner): preprocessing called on
+                complex inputs. Note that this combiner must also accept
+                `input_tensor_spec` as the input to compute the processed
+                tensor spec. For example, see `alf.nest.utils.NestConcat`. This
+                arg is helpful if you want to combine inputs by configuring a
+                gin file without changing the code.
+            conv_layer_params (tuple[tuple]): a tuple of tuples where each
+                tuple takes a format `(filters, kernel_size, strides, padding)`,
+                where `padding` is optional.
+            fc_layer_params (tuple[int]): a tuple of integers representing hidden
+                FC layer sizes.
+            lstm_hidden_size (int or tuple[int]): the hidden size(s)
+                of the LSTM cell(s). Each size corresponds to a cell. If there
+                are multiple sizes, then lstm cells are stacked.
+            actor_fc_layer_params (tuple[int]): a tuple of integers representing
+                hidden FC layers that are applied after the lstm cell's output.
+            activation (nn.functional): activation used for hidden layers. The
+                last layer will not be activated.
+            name (str): name of the network
+        """
+        super(ActorRNNNetwork, self).__init__(
+            input_tensor_spec,
+            input_preprocessors,
+            preprocessing_combiner,
+            name=name)
+
+        self._action_spec = action_spec
+        flat_action_spec = nest.flatten(action_spec)
+        self._flat_action_spec = flat_action_spec
+
+        is_continuous = [
+            single_action_spec.is_continuous
+            for single_action_spec in flat_action_spec
+        ]
+
+        assert all(is_continuous), "only continuous action is supported"
+
+        self._encoding_net = EncodingNetwork(
+            input_tensor_spec=self._processed_input_tensor_spec,
+            conv_layer_params=conv_layer_params,
+            fc_layer_params=fc_layer_params,
+            activation=activation)
+
+        self._lstm_encoding_net = LSTMEncodingNetwork(
+            input_tensor_spec=self._encoding_net.output_spec,
+            hidden_size=lstm_hidden_size,
+            post_fc_layer_params=actor_fc_layer_params,
+            activation=activation)
```

Comment:
Can just combine the two encoding networks into one LSTMEncodingNetwork

Reply:
merged into LSTMEncodingNetwork